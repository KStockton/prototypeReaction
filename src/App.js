import React, { Component } from 'react'
import './App.scss'
import Card from './Card'
import Title from './Title'


export default class App extends Component {
  constructor(){
    super()
    this.state = {
      prototypes: [],
      topicChoice: null,
      studyCards: [],
      review: false,
    }
  }

  componentDidMount() {
    fetch('https://fe-apps.herokuapp.com/api/v1/memoize/1901/michaelks-datset/learnprototypes')
      .then(response => response.json())
      .then(prototypes => this.setState({ 
        prototypes: prototypes }))
      .catch(error => console.log(error))
  }
compon
  
  componentWillUpdate(){
    if(this.state.studyCards.length === localStorage.length) {
      return true;
    } else {
    for(let i = 0; i < localStorage.length; i++) {
      console.log('localStorage', localStorage)
     this.state.studyCards.push(JSON.parse(localStorage.getItem(localStorage.key(i))))
    }
  }
  }
  
    
    selectTopic = (topic) => {
      let topicChoice = this.state.prototypes.learnPrototypes.filter(prototype => {   
        return prototype.category === topic
      })
    this.setState({topicChoice: topicChoice})
      }

      selectReview = () => {
        this.setState({ review: !this.state.review})
      }

      
      render() {
        console.log(this.state.review)
      //  console.log('this.state.prototypes', this.state.prototypes)
        console.log('studyCards', this.state.studyCards)
        let studyMode
        if(this.state.review  === true){
          studyMode = <Card topicChoice={this.state.studyCards}/>
        } else {
          studyMode = <Card topicChoice={this.state.topicChoice}/>
        }
        return (
          <div className="App">
        <header className='App-header'>   
          <h1>Want to Play a Game</h1>
        </header>
        <Title selectTopic={this.selectTopic} selectReview={this.selectReview}/>
        {studyMode}
      </div>
    )
  }
}