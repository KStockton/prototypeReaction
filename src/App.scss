
.App {
  text-align: center;
  margin: auto;
  box-sizing: border-box;
}

.App-header {
  display: flex;
  justify-content: center;
  align-items: center;
  color: #ffffff;
  padding: 20px;
  margin: auto;
  font-size: 30px;
}

h1 {
  font-weight: 300;
  border-bottom: 3px solid white;
} 


