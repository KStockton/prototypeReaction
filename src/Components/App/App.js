import React, { Component } from 'react'
// import './_App.scss'
import Card from '../Card/Card';
import Title from '../Title/Title';

export default class App extends Component {
  constructor(){
    super()
    this.state = {
      prototypes: [],
      topicChoice: null,
      studyCards: [],
      review: false,
      localStorage: 0,
    }
  };

  componentDidMount() {
    fetch('https://fe-apps.herokuapp.com/api/v1/memoize/1901/michaelks-datset/learnprototypes')
      .then(response => response.json())
      .then(prototypes => this.setState({ 
        prototypes: prototypes.learnPrototypes }))
      .catch(error => console.log(error))
  };
  
  componentWillUpdate(){ 
    
    for(let i = 0; i < localStorage.length; i++) {
     this.state.studyCards.push(JSON.parse(localStorage.getItem(localStorage.key(i))))
    }
  };
  
    
    selectTopic = (topic) => {
      let topicChoice = this.state.prototypes.filter(prototype => {   
        return prototype.category === topic
      })
      this.setState({topicChoice})
    };

      selectReview = () => {
        this.setState({ review: !this.state.review})
      };
      
      render() {
        const { review, studyCards, topicChoice } = this.state;
        let studyMode;

        if(review  === true && studyCards.length > 0){
          studyMode = <Card topicChoice={studyCards}/>
        } else {
      
          studyMode = <Card topicChoice={topicChoice}/>
        };

        return (
          <div className="App">
            <header className='App-header'>   
            <h1>Prototype Building Blocks</h1>
            </header>
          <Title selectTopic={this.selectTopic} selectReview={this.selectReview} review={studyCards}/>
            {studyMode}
          </div>
    )
  }
};