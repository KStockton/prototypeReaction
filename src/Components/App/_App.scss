@import '../../Sass/variables.scss'; 

  body {
    background-image: url('../assets/memobg.jpg');
    background-size: 100% 100%;
    background-repeat: no-repeat;
    margin: 0;
    padding: 0;
    font-family: 'Exo', sans-serif;
  }

.App {
  text-align: center;
  margin: auto;
  box-sizing: border-box;
}

.App-header {
  @include displayFlex(center, center, null, null);
  color: $border-color;
  padding: 20px;
  margin: auto;
  font-size: 30px;
  font-family: 'Black Ops One', cursive;
}

h1 {
  font-weight: 300;
  border-bottom: 3px solid $border-color;
} 


