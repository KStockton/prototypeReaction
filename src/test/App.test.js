import React from 'react';
import App from '../App';
import { shallow } from "enzyme";
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() })

describe('App', () => {
  let wrapper;

  beforeEach(() => {
   wrapper = shallow(<App />
   )
  });

  it('should match the snapshot with all data passed in', () =>{
   expect(wrapper).toMatchSnapshot()
  })

})