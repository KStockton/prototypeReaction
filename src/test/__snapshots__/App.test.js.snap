// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot with all data passed in 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <h1>
      Want to Play a Game
    </h1>
  </header>
  <Title
    selectReview={[Function]}
    selectTopic={[Function]}
  />
  <Card
    topicChoice={null}
  />
</div>
`;
